prism.js

```diff
@@ -1537,7 +1537,7 @@ Prism.languages.javascript = Prism.languages.extend('clike', {
 	],
 	// Allow for all non-ASCII characters (See http://stackoverflow.com/a/2008444)
 	'function': /#?(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\s*(?:\.\s*(?:apply|bind|call)\s*)?\()/,
-	'number': /\b(?:(?:0[xX](?:[\dA-Fa-f](?:_[\dA-Fa-f])?)+|0[bB](?:[01](?:_[01])?)+|0[oO](?:[0-7](?:_[0-7])?)+)n?|(?:\d(?:_\d)?)+n|NaN|Infinity)\b|(?:\b(?:\d(?:_\d)?)+\.?(?:\d(?:_\d)?)*|\B\.(?:\d(?:_\d)?)+)(?:[Ee][+-]?(?:\d(?:_\d)?)+)?/,
+	'number': /\b(?:(?:0[xX](?:[\dA-Fa-f](?:_[\dA-Fa-f])*)+|0[bB](?:[01](?:_[01])*)+|0[oO](?:[0-7](?:_[0-7])*)+)n?|(?:\d(?:_\d)*)+n|NaN|Infinity)\b|(?:\b(?:\d(?:_\d)*)+\.?(?:\d(?:_\d)*)*|\B\.(?:\d(?:_\d)*)+)(?:[Ee][+-]?(?:\d(?:_\d)*)+)?/,
```

Comment:
> Bug is that number in a variable starting with '$' gets matched.

You're right. Here's a fix.

```suggestion
	'number': { 
		pattern: /(^|[^\w$])(?:(?:0[xX](?:[\dA-Fa-f](?:_[\dA-Fa-f])*)+|0[bB](?:[01](?:_[01])*)+|0[oO](?:[0-7](?:_[0-7])*)+)n?|(?:\d(?:_\d)*)+n|NaN|Infinity)\b|(?:\b(?:\d(?:_\d)*)+\.?(?:\d(?:_\d)*)*|\B\.(?:\d(?:_\d)*)+)(?:[Ee][+-]?(?:\d(?:_\d)*)+)?/,
		lookbehind: true
	},
```

Reply:
That doesn't work

Reply:
This should work: 
`'number': /\b(?:(?:0[xX](?:[\dA-Fa-f](?:_[\dA-Fa-f])*)+|0[bB](?:[01](?:_[01])*)+|0[oO](?:[0-7](?:_[0-7])*)+)n?|(?:\d(?:_\d)*)+n|NaN|Infinity)\b|(?:\b(?:\d(?:_\d)*)+\.?(?:\d(?:_\d)*)*|\B\.(?:\d(?:_\d)*)+)(?:[Ee][+-]?(?:\d(?:_\d)*)+)?/,
`

Reply:
I updated the suggestion. This time, everything should work.